src/mango/src/mango_cursor_nouveau.erl

```diff
@@ -171,26 +173,39 @@ handle_hit(CAcc0, Hit, Doc) ->
     #cacc{
         limit = Limit,
         skip = Skip,
-        execution_stats = Stats
+        execution_stats = Stats,
+        documents_seen = Seen
     } = CAcc0,
     CAcc1 = update_bookmark(CAcc0, Hit),
     Stats1 = mango_execution_stats:incr_docs_examined(Stats),
     couch_stats:increment_counter([mango, docs_examined]),
     CAcc2 = CAcc1#cacc{execution_stats = Stats1},
     case mango_selector:match(CAcc2#cacc.selector, Doc) of
-        true when Skip > 0 ->
-            CAcc2#cacc{skip = Skip - 1};
-        true when Limit == 0 ->
-            % We hit this case if the user spcified with a
-            % zero limit. Notice that in this case we need
-            % to return the bookmark from before this match
-            throw({stop, CAcc0});
-        true when Limit == 1 ->
-            NewCAcc = apply_user_fun(CAcc2, Doc),
-            throw({stop, NewCAcc});
-        true when Limit > 1 ->
-            NewCAcc = apply_user_fun(CAcc2, Doc),
-            NewCAcc#cacc{limit = Limit - 1};
+        true ->
+            DocId = mango_fields:extract(Doc, [<<"_id">>]),
```

Comment:
This looks like won't give us the `DocId`, but a whole document object with an `_id` field and value? It seems `documents_seen` could just be a simple set of binary DocIds like `<<"doc1">>, <<"doc2">>, ...`

Reply:
Oh, that is a nice catch!  My attention was probably carried away by the regular field extraction logic.  Yes, you are absolutely right, this could be much simpler.